app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java

```diff
@@ -108,6 +113,39 @@ public void onCreatePreferences(Bundle bundle, String rootKey) {
             return true;
         });
 
+        ListPreference countryPreference = ((ListPreference) findPreference("user_country"));
+        List<String> countryLabels = new ArrayList<>();
+
+        DaoSession daoSession = OFFApplication.getInstance().getDaoSession();
+        AsyncSession asyncSessionCountries = daoSession.startAsyncSession();
+        CountryNameDao countryNameDao = daoSession.getCountryNameDao();
+
+        asyncSessionCountries.queryList(countryNameDao.queryBuilder()
+                .where(CountryNameDao.Properties.LanguageCode.eq(LocaleHelper.getLanguage(getActivity())))
+                .orderAsc(CountryNameDao.Properties.Name).build());
+
+        asyncSessionCountries.setListenerMainThread(operation -> {
```

Comment:
You should set the listener before querying.

Reply:
Yes, I'll fix this.

Reply:
This has been fixed.